MusicTaste/MusicTaste/View Controllers/MatchmakingViewController.m

```diff
@@ -74,16 +77,16 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 -(void)getMatchesDictionary{
     PFQuery *music = [PFQuery queryWithClassName:@"Music"];
     NSMutableArray *allMatchesArray = [self.author valueForKey:@"matches"];
-    NSMutableDictionary *matchesDict = [allMatchesArray objectAtIndex:0];
-    self.matchesObjectIds = [matchesDict keysSortedByValueUsingSelector:@selector(compare:)];
     //query through each match
     [music findObjectsInBackgroundWithBlock:^(NSArray *users, NSError *error) {
         if (users != nil) {
-            self.matches = [[NSMutableArray alloc] initWithArray:users];
-            for (PFObject *user in users){
-                if (![self.matchesObjectIds containsObject: user.objectId]){
-                    [self.matches removeObject:user];
-                    [self.MatchTableView reloadData];
+            self.matches = [[NSMutableArray alloc] init];
+            for (NSString *matchid in allMatchesArray){
+                for (PFObject *user in users){
+                    if ([matchid isEqual: user.objectId]){
+                        [self.matches addObject:user];
+                        [self.MatchTableView reloadData];
+                    }
```

Comment:
Any concerns about time complexity here? Seems like `O(m * u)` (matches times users). If you have a lot of matches then it could approach `O(u^2)`. Are there any protections in place to limit, or can you use an approach to make finding the PFObject constant time?

Reply:
I can limit the amount of matches so it is constant but I do see your concern.